src/App.tsx

```diff
@@ -1,36 +1,45 @@
-import { Flex, useDisclosure, Text, Box } from '@chakra-ui/react'
+import { Box, Flex, Text, useDisclosure } from '@chakra-ui/react'
 import Overview from './views/Overview'
 import { ViewState } from './constants/enums'
 import { useApplicationContext } from './contexts/applicationContext'
 import Macroview from './views/Macroview'
 import { useEffect } from 'react'
-import SettingsModal from './views/SettingsModal'
 import data from '@emoji-mart/data'
 import { init } from 'emoji-mart'
 import './App.css'
 import { MacroProvider } from './contexts/macroContext'
+import MacroSettingsModal from './views/MacroSettingsModal'
+import SettingsModal from './views/SettingsModal'
 
 function App() {
   const { viewState, initComplete } = useApplicationContext()
-  const { isOpen, onOpen, onClose } = useDisclosure()
+  const {
+    isOpen: isOpenSettings,
+    onOpen: onOpenSettings,
+    onClose: onCloseSettings
+  } = useDisclosure()
+  const {
+    isOpen: isOpenMacroSettings,
+    onOpen: onOpenMacroSettings,
+    onClose: onCloseMacroSettings
+  } = useDisclosure()
 
   useEffect(() => {
-    document.addEventListener('contextmenu', (event) => event.preventDefault()) // disables tauri's right click context menu
-    // TODO: Figure out how to disable other keyboard shortcuts like CTRL-F
+    document.addEventListener('contextmenu', (event) => event.preventDefault()) // disables Tauri right click context menu
+    // TODO: Add disable for ctrl+r and f5, but only when in debug mode - envvar RUST_LOG='debug'
+    document.addEventListener('keydown', (event) => {
+      if (event.ctrlKey && event.key.toLowerCase() === 'f') {
+        event.preventDefault()
+      }
+    })
```

Comment:
What's the reasoning behind disabling Ctrl+F? 
Also, won't this prevent binding macros to shortcuts that contain Ctrl & F? As the macro binding logic uses the keydown function

Reply:
Idea was to prevent the webpage search bar from appearing, the macro binding still works (just tested)

Reply:
It would be good to add that context as a comment then

Reply:
Played around with the concept a bit better, tried this: [f80d476](https://github.com/WootingKb/wooting-macros/pull/194/commits/f80d476a0706253f4164e241766865ef35ef95bd)